internal/controller/pkg/revision/reconciler.go

```diff
@@ -281,8 +301,11 @@ func SetupProviderRevision(mgr ctrl.Manager, o controller.Options) error {
 		Watches(&v1alpha1.ControllerConfig{}, &EnqueueRequestForReferencingProviderRevisions{
 			client: mgr.GetClient(),
 		}).
+		Watches(&v1beta1.DeploymentRuntimeConfig{}, &EnqueueRequestForReferencingProviderRevisions{
+			client: mgr.GetClient(),
+		}).
```

Comment:
Yes, it should be safe since we always introduce the type.

Despite being safe, it was unnecessary, but I didn't feel motivated to change the code there as it would look different than how we used to. Now, you're noticing it, I did it properly: https://github.com/crossplane/crossplane/pull/4744/commits/ecd2b0bca2d4a56e7dd14e3acf8670cf81c896eb